### PR TITLE
Add Guard clause against contract creation from non verified Adresses and not execute function that are blacklisted

### PR DESCRIPTION
**Pull request by DerTiedemann**
- Enable admin enforcement
- Make all backends aware of the change && enable on main backend
- Fix old tests
- enforcement tests

This PR introduces a simple guard clause inside of the CREATE instruction that checks if the caller address has the KYC flag set in the precompiled contract at `0x010000000000000000000000000000000000000a`. This SC is deployed as one of the precompiled SCs.

Furthermore I added the test ethaddress (the one thats funded) to be the admin for this admin contract to be able to add/remove roles for address for testing.
Now when an address thats not verified tries to create a contract:

- The initial test will fail and return an error
- if that error is ignored the TX will fail

Also a Blacklist is introduced that will make any tx fail that in any way or form call a function in that list.
Everywhere were the NativeAssetCall is blocked by avax i added a check if the function signature is present in the admin contract and return an error.
Currently this only causes the TX to fail, e.g. it costs gas and does not revert.
Setting the functionselctor `0x00000000` on a contract will disable all of its functions e.g. disable it to be called alltogether.

For easier testing, this whole behavior is controlled by a new value in the vm.Config struct called EnableAdminEnforcement. This allowes the VM to act like normal and do not break test. Be aware that this has to be turned on explicitly.

All the revelant places have been updated to be aware of the changes and the chain will boot with enforcement enabled.

### CODE REVIEW

File: accounts/abi/bind/backends/simulated.go [hunk 1]
```diff
@@ -131,6 +131,32 @@ func NewSimulatedBackendWithDatabase(database ethdb.Database, alloc core.Genesis
 	return backend
 }
 
+func NewSimulatedBackendWithDatabaseAndValidKYCRole(database ethdb.Database, alloc core.GenesisAlloc, gasLimit uint64, addr common.Address) *SimulatedBackend {
```
Comment (c4t-ag) [inline accounts/abi/bind/backends/simulated.go:134]: They should go in its own camino_simulated.go file.
Side node: If we change foreign code we adjust license header

File: core/admin/admin_interface.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
+package admin
```
Comment (c4t-ag) [inline core/admin/admin_interface.go:1]: License Header

File: core/admin/admin_interface_test.go [hunk 3]
```diff
@@ -0,0 +1,203 @@
+package admin_test
```
Comment (c4t-ag) [inline core/admin/admin_interface_test.go:1]: License header

**Review by c4t-ag — CHANGES_REQUESTED** [review]
Needs clarification before continuing / fixing

File: core/admin/admin_interface.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+	FucntionSelectorLength = 4
+)
+
+var (
+	DisableContractFunctionSelector = common.FromHex("0x00000000")
+)
+
+func KycRoleKeyFromAddr(addr common.Address) common.Hash {
+	return crypto.Keccak256Hash(append(addr.Hash().Bytes(), common.HexToHash("0x2").Bytes()... /*slot 2 reference admin.sol map(address => uint)*/))
```
Comment (pnowosie) [inline core/admin/admin_interface.go:51]:
```suggestion
	return crypto.Keccak256Hash(addr.Hash().Bytes(), common.HexToHash("0x2").Bytes()/*slot 2 reference admin.sol map(address => uint)*/)
```
you can drop `append` which leads the same result

**Comment by DerTiedemann** [comment]
Implemented by the Admin controller

**Pull request closed without merging**